### PR TITLE
[AIR] [Docs] Update "Training a Torch classifier"

### LINKED ISSUE #28159: [AIR] Update "Training a Torch classifier" example

### PR DESCRIPTION
**Pull request by bveeramani**
Signed-off-by: Balaji Veeramani [balaji@anyscale.com](mailto:balaji@anyscale.com)

## Why are these changes needed?

1. The example wasn't using `TorchCheckpoint`
2. The example was using `iter_torch_batches` incorrectly
3. The example overcomplicated serving

**Depends on:**
- [x] #27970 

## Related issue number

Closes #28159

## Checks

- [ ] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [x] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [x] This PR is not tested :(

### CODE REVIEW

File: doc/source/ray-air/examples/torch_image_example.ipynb [hunk 1]
```diff
@@ -7,21 +7,90 @@
...
    "execution_count": 1,
    "id": "d806ba6b",
    "metadata": {},
-   "outputs": [],
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "Requirement already satisfied: ray[air] in /Users/bveeramani/GitHub/ray/python (3.0.0.dev0)\n",
```
Comment (amogkam) [inline doc/source/ray-air/examples/torch_image_example.ipynb:29]: Can we cleanup this output?

File: doc/source/ray-air/examples/torch_image_example.ipynb [hunk 2]
```diff
@@ -140,14 +241,14 @@
    "id": "a89b59e8",
    "metadata": {},
    "source": [
-    "Note that {py:class}`SimpleTorchDatasource <ray.data.datasource.SimpleTorchDatasource>` loads all data into memory, so you shouldn't use it with larger datasets.\n",
+    "{py:class}`SimpleTorchDatasource <ray.data.datasource.SimpleTorchDatasource>` is unperformant, so you shouldn't use it with larger datasets.\n",
```
Comment (amogkam) [inline doc/source/ray-air/examples/torch_image_example.ipynb:244]: Hmm the sentence you had previously seems more clear to me.

File: doc/source/ray-air/examples/torch_image_example.ipynb [hunk 3]
```diff
@@ -403,34 +622,51 @@
...
+      "\u001b[2m\u001b[36m(BlockWorker pid=5938)\u001b[0m A value is trying to be set on a copy of a slice from a DataFrame.\n",
+      "\u001b[2m\u001b[36m(BlockWorker pid=5938)\u001b[0m Try using .loc[row_indexer,col_indexer] = value instead\n",
+      "\u001b[2m\u001b[36m(BlockWorker pid=5938)\u001b[0m \n",
+      "\u001b[2m\u001b[36m(BlockWorker pid=5938)\u001b[0m See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
+      "\u001b[2m\u001b[36m(BlockWorker pid=5938)\u001b[0m   df.loc[:, col_name] = TensorArray(col)\n",
+      "\u001b[2m\u001b[36m(BlockWorker pid=5938)\u001b[0m /Users/bveeramani/GitHub/ray/python/ray/air/util/data_batch_conversion.py:158: SettingWithCopyWarning: \n",
+      "\u001b[2m\u001b[36m(BlockWorker pid=5938)\u001b[0m A value is trying to be set on a copy of a slice from a DataFrame.\n",
+      "\u001b[2m\u001b[36m(BlockWorker pid=5938)\u001b[0m Try using .loc[row_indexer,col_indexer] = value instead\n",
```
Comment (amogkam) [inline doc/source/ray-air/examples/torch_image_example.ipynb:647]: can we remove these warnings?

**Review by amogkam — COMMENTED** [review]
Code changes lgtm, thanks! But let's clean up some of the output in the notebook.

**Review by amogkam — APPROVED** [review]

**Review by richardliaw — APPROVED** [review]
stamp

**Pull request merged**